modules/io/src/writers/xviz-protobuf-writer.js

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/* eslint-disable */
+import {XVIZBaseWriter} from './xviz-base-writer';
+import {XVIZ_PROTOBUF_MAGIC, XVIZ_PROTOBUF_MESSAGE} from '../common/protobuf-support';
+
+// 0-frame is an index file for timestamp metadata
+// 1-frame is the metadata file for the log
+// 2-frame is where the actual XVIZ updates begin
+const messageName = index => `${index + 2}-frame`;
+
+export class XVIZProtobufWriter extends XVIZBaseWriter {
```

Comment:
Is the message timing bits here protobuf specific, or could it be in the base class?

Reply:
You are correct, it is shared. But I will address this in a separate PR